### PR TITLE
OLS-244: Prompt generator

### PR DESCRIPTION
**Pull request by tisnik**
## Description

[OLS-244](https://issues.redhat.com//browse/OLS-244): Prompt generator

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #[OLS-244](https://issues.redhat.com//browse/OLS-244)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Review by asamal4 — COMMENTED** [review]
As a first step for this work, i think basic flow looks good with default prompt. 
Please add TODOs for core logic implementation for dynamic prompt modification.

File: ols/src/prompts/prompt_generator.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    rag_exists: bool,
+    history_exists: bool,
+    default_prompt: ChatPromptTemplate,
+) -> ChatPromptTemplate:
+    """Find prompt for given configuration parameters or return the default one."""
+    selector = PromptConfiguration(provider, model, rag_exists, history_exists)
+    return prompt_configurations.get(selector, default_prompt)
+
+
+def generate_prompt(
```
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/src/prompts/prompt_generator.py:40]: yes, I'm on it.

File: ols/src/prompts/prompt_generator.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+from .prompts import CHAT_PROMPT
+
+PromptConfiguration = namedtuple(
+    "PromptConfiguration", "provider model rag_exists history_exists"
+)
+
+prompt_configurations = {
+    PromptConfiguration(PROVIDER_BAM, GRANITE_13B_CHAT_V1, True, True): CHAT_PROMPT,
```
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/src/prompts/prompt_generator.py:23]: I will add TODO accordingly

File: ols/src/query_helpers/docs_summarizer.py [hunk 3]
```diff
@@ -135,11 +145,22 @@ def summarize(
 
         rag_context, referenced_documents = self._format_rag_data(rag_context_data)
 
-        interim_prompt = CHAT_PROMPT.format(
-            context=rag_context, query=query, chat_history=[]
+        provider_config = config.llm_config.providers.get(self.provider)
+
+        model_options = self._get_model_options(provider_config)
+
+        interim_prompt = generate_prompt(
```
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/src/query_helpers/docs_summarizer.py]: implemented, thank you @asamal4

File: ols/src/prompts/prompt_generator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/src/prompts/prompt_generator.py:41]: added some tests @asamal4

**Review by asamal4 — COMMENTED** [review]
some minor typos.

File: ols/src/prompts/prompt_generator.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+"""Prompt generator based on context."""
```
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/src/prompts/prompt_generator.py]: fixed, thank you

File: tests/unit/prompts/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+"""Unit tests for promt generator."""
```
[the remark this answers is not in the document]
Comment (tisnik) [reply tests/unit/prompts/__init__.py]: fixed, thank you

File: tests/unit/prompts/test_prompt_generator.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+@pytest.mark.parametrize("query", queries)
+def test_generate_prompt_default_prompt(provider, model, query):
+    """Test if prompt generator returns default prompt for given input."""
+    model_options = {}
+    conversation_id = "154e0444-db72-44fd-a6be-4ba0c15059a4"
+    history = []
+    rag_context = ""
+    referenced_documents = []
+
+    # currently, defalt CHAT_PROMPT should be returned for all cases
```
[the remark this answers is not in the document]
Comment (tisnik) [reply tests/unit/prompts/test_prompt_generator.py]: fixed, thank you

**Pull request merged**